runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -941,14 +958,18 @@ public void addWorkerStatusPage(BaseStatusServlet page) {
   public void stop() {
     try {
       if (globalConfigRefreshTimer != null) {
-        globalConfigRefreshTimer.cancel();
+        globalConfigRefreshTimer.shutdown();
+        globalConfigRefreshTimer.awaitTermination(300, TimeUnit.SECONDS);
```

Comment:
nit: the previous semantics of cancel() are unclear if the task is running, it may not block if it was running just prevent it from running again.  In that case you may be increasing shutdown time here by stopping things serially.  You could instead cancel everything and then await on them all.

Reply:
Done. The code now shuts down the ScheduledExecutorService instances first, then awaits termination on all of them.

What do you think about calling .shutdownNow() instead of .shutdown()? This would send an interrupt to any Runnable that's currently executing.

Reply:
It's really only in tests, so I think if it was a concern we would have timeouts already.
I'd be hesitant to change it just in case it makes something flaky if it isn't currently a problem.
Could leave a comment about that being possible to speed up shutdown.

Reply:
Left a comment.